extended-utxo-spec/extended-utxo-specification.tex

```diff
@@ -0,0 +1,1212 @@
+redeemer as input, and if it terminates without an error arising then
+the output can be spent.  Before a transaction can proceed, all inputs
+must be successfully validated: if one or more inputs fails to validate
+then the transaction is rejected (\blue{is this true?}).
+
+A simple example of this is a \i{pay-to-pubkey} script, where the
+redeemer consists of a signature for the current transaction produced
+using a private key belonging to the owner of the output.  The
+validator (provided by the owner of the output) would check the
+signature using a known public key: if the public key corresponds to
+the private key then validation succeeds, otherwise it fails.  Thus
+the output can only be spent by the owner of the relevant private key
+
+\paragraph{Blockchain terminology.}
+\label{sec:blockchain-terminology}
+We require a little blockchain background. A \textit{blockchain} is a
+ledger consisting of a sequence of \textit{blocks}, each block
+containing a number of transactions (and possibly other data).  A
+typical blockchain will be maintained by a peer-to-peer network
+consisting of \textit{core nodes} who compete to execute users'
+transactions and create new blocks, generally receiving some monetary
+reward for doing so (for example, in order to have a transaction
+executed on the chain, users may have to pay a \textit{fee} calculated
+from the transaction size and execution time). After a new block has
+been created, it will be propagated to the rest of the network, which
+will eventually reach consensus on the current state of the
+blockchain.  The opportunity to create a new block is known as a
+\textit{slot}: slots occur at regular intervals (about once every 20
+seconds on Cardano), and hence the current \textit{slot number} can be
+used as a proxy for time on the blockchain (using real times would
+lead to problems with accuracy and synchronisation).
+
+
+\section{XUTXO-1: Enhanced scripting}
+\label{sec:xutxo-1}
+The XUTXO-1 model adds the following new features to the model 
+proposed in~\cite{Zahnentferner18-UTxO}:
+
+\begin{itemize}
+\item Every transaction has a \textit{validity interval}, consisting
+  of two slot numbers: a core node will only process the transaction
+  if the current slot number lies within the transaction's validity
+  interval.
+
+\item We introduce a new type of script called a \i{data script}, and
+  each unspent output has a (possibly empty) data script associated
+  with it.  The introduction of the data script increases the
+  expressivity of the model considerably. For example, one can use the
+  datascript to propagate state between transactions, and this can be
+  used to give a contract the structure of a finite state machine.
+
+\item During validation, validator scripts are provided with
+  information about the current state of the blockchain.  This
+  includes things such as the current slot number, the validity
+  interval of the current transaction, and the hash of the
```

Comment:
Thanks: that was a silly mistake.  This does raise the question of exactly what we can put in $P$.  All of the rest of the document makes sense whatever P contains, but validation being deterministic depends on ... what exactly?  The contents of P must be the same whatever the state of the blockchain (and maybe the outside world) is?

Reply:
The value of P should be the same for all possible evolutions of the chain in which this transaction is validated.

Which basically says it can have information about the transaction itself and not much else.